### ISSUE TITLE
upsert the index when using the ToDatabaseScript.

### ISSUE THREAD

**Issue by dbones** [JasperFx/marten#2072]
can we change the

`CREATE UNIQUE INDEX`  to `CREATE UNIQUE INDEX IF NOT EXISTS`

we are currently using Marten to generate the SQL and then apply it to Postgres, with a DDL Role applied (this is a quick workaround to assign the desired Role against the created schema and tables, which is not the same role our application connects with, I believe this is similar to issue #1474)

```
//this is the role that will be the owner of all things in the database schema
options.Advanced.DdlRules.Role = "application-owner"; //application connects with another role, which is a member of this.
```

At the start of the application we apply the script generated by the `ToDatabaseScript` method, similar to the following:

```
using IDbConnection connection = new NpgsqlConnection("connectionString");
connection.Open();

using var trx = connection.BeginTransaction();
using var command = connection.CreateCommand();

command.Transaction = trx;
command.CommandType = CommandType.Text;

var dbScript = store.Schema.ToDatabaseScript();
command.CommandText = dbScript;
command.ExecuteNonQuery();

trx.Commit();
```
 
 The index is created by the application running the first time, it will fail the second run, as the index has been created.

**Comment by jeremydmiller** [comment]
@dbones To be honest, I'm really wanting to ditch the DDL template stuff in Marten V5 and just let folks export and modify the DDL scripts by hand. The usage is very rare I believe, and it's extra complexity in Marten I'd love to eliminate. 

But talk to me here, why is Marten even trying to recreate that index?

**Comment by dbones** [comment]
@jeremydmiller 3 things

**for this issue**

I do not want to recreate them, only ensure that they are created (and updated)

As the application starts, it will run the code above (which sets up the database)

If you restart it or run another instance, the DDL, command fails, as the index is already there.

**why we employed this workaround**

I do not think this is an issue if we get marten to auto-handle this. 

however, when we used marten to handle the creations of schema/tables/functions directly (auto-create = true). It was created with the owner of the applications login account, not with a role in which the login account is a member of 

```
roles
[application-login] --- member of -----> [application-owner]
```

**also, can I clarify the scope of your statement?**

does this mean marten will not auto-create database/schema/tables/functions/indexes? (i.e. we will need to employ some migrations library like https://fluentmigrator.github.io/ to generate all the postgres artifacts)

(if it helps) One massive advantage for us is that it handles this, it allows us to think of Postgres more like Couchdb or Mongo, without us having to adopt new technologies. (it's exceptional and I have been able to get people to rally around it)
